include/boost/url/any_allocator.hpp

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright (c) 2022 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_ANY_ALLOCATOR_HPP
+#define BOOST_URL_ANY_ALLOCATOR_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/detail/any_allocator.hpp>
+
+#include <memory>
+#include <type_traits>
+
+namespace boost {
+namespace urls {
+
+/** A type-erased allocator with shared ownership.
+
+    This type satisfies the requirements for <em>Allocator</em>
+
+    @par Specification
+    @li <a href="https://en.cppreference.com/w/cpp/named_req/Allocator"
+       >Allocator (cppreference.com)</a>
+
+ */
+#ifdef BOOST_URL_DOCS
+template< class T >
+using any_allocator = __see_below __;
+#else
+template<class T>
+class any_allocator
+    : private detail::any_allocator_base
+{
+    std::shared_ptr<base> p_;
+
+    template<class U>
+    friend class any_allocator;
+
+public:
+    using value_type = T;
+
+    template<class U>
+    struct rebind
+    {
+        using other = any_allocator<U>;
+    };
+
+    using is_always_equal = std::false_type;
+
+    any_allocator() noexcept;
+
+    any_allocator(any_allocator const&) noexcept = default;
```

Comment:
Does this need to say `noexcept`?

Reply:
The constructor needs it.

![image](https://user-images.githubusercontent.com/5369819/149649390-45cff300-df4d-4a03-ac5a-c8cdeae9a0ee.png)

Reply:
Yes but the default copy constructor for `any_allocator` is implicitly noexcept